File: src/main/java/com/ject/studytrip/mission/infra/jpa/DailyMissionRepositoryAdapter.java
package com.ject.studytrip.mission.infra.jpa;

import com.ject.studytrip.mission.domain.model.DailyMission;
import com.ject.studytrip.mission.domain.repository.DailyMissionRepository;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

@Repository
@RequiredArgsConstructor
public class DailyMissionRepositoryAdapter implements DailyMissionRepository {
    private final DailyMissionJpaRepository dailyMissionJpaRepository;

    @Override
    public DailyMission save(DailyMission dailyMission) {
        return dailyMissionJpaRepository.save(dailyMission);
    }

    @Override
    public List<DailyMission> saveAll(List<DailyMission> dailyMissions) {
        return dailyMissionJpaRepository.saveAll(dailyMissions);
    }

    @Override
    public List<DailyMission> findAllByIdIn(List<Long> ids) {
        return dailyMissionJpaRepository.findAllByIdIn(ids);
    }
}

File: src/main/kotlin/com/ject/studytrip/mission/infra/jpa/DailyMissionRepositoryAdapter.kt
package com.ject.studytrip.mission.infra.jpa

import com.ject.studytrip.mission.domain.model.DailyMission
import com.ject.studytrip.mission.domain.repository.DailyMissionRepository
import org.springframework.stereotype.Repository

@Repository
class DailyMissionRepositoryAdapter(
    private val dailyMissionJpaRepository: DailyMissionJpaRepository,
) : DailyMissionRepository {
    override fun save(dailyMission: DailyMission): DailyMission = dailyMissionJpaRepository.save(dailyMission)

    override fun saveAll(dailyMissions: List<DailyMission>): List<DailyMission> = dailyMissionJpaRepository.saveAll(dailyMissions)

    override fun findAllByIdIn(dailyMissionIds: List<Long>): List<DailyMission> = dailyMissionJpaRepository.findAllByIdIn(dailyMissionIds)
}
